### ISSUE TITLE
Syntax of if-else-loop condition in Bayesian networks

### ISSUE THREAD

**Issue by SEICS** [SHoltzen/dice#58]
Hi,

Although I have looked into some examples and tried writing some simple Bayesian networks, I was really confused by the if-loop condition in dice. 

In the examples given, I have already seen 3 ways to define the if-loop conditions for Bayesian networks:

1. In alarm.bif.dice, the syntax for defining PAP. 
2. In alarm.bif.dice, the syntax for defining VENTMACH. 
3. In hailfinder.bif.dice, the syntax for defining ScenRelAMIns.

To me, it seems like dice doesn't have a consistent syntax for if-loop conditions when parents have different lengths in Bayesian networks.

I would like to know **what is the proper syntax for defining children in a Bayesian network** and what int(int1, int2) stands for. In the paper and instructions given in GitHub, int(int1, int2) should be the values between int1 and int2. However, when I tried to define a Bayesian network (A-->B) where A has 7 values from 0 to 6 and B has 2 values from 0 to 1, I defined the condition for judging A as int(1,0) for A=0, int(2,1) for A=1, and so on but dice threw errors and said it should be int(3,0) for A=0, etc. 

Can anyone give some explanations on it? Thank you so much!

**Comment by SEICS** [comment]
Just to update that, I guessed that for if-loop condition expression "let y = if (x == int(int1, int2)) then ...", int2 will be the values/states that y's parent could have. However, I still cannot figure out how to calculate int1. At first, I thought it could be the integer division of the number of parent states divided by two, and I have also tried to substitute parents states to children states but none of them really apply to the scheme. 

Can anyone help me with this? Thank you!
